File: src/content/docs/topics/cs/coding-problems/stack/907-sum-of-subarray-minimums.py
def sum_subarray_mins(arr: list[int]) -> int:
    pass  # TODO: implement

def _run_tests():
    assert sum_subarray_mins([3, 1, 2, 4]) == 17
    assert sum_subarray_mins([11, 81, 94, 43, 3]) == 444
    assert sum_subarray_mins([3]) == 3
    print("all tests pass")

if __name__ == "__main__":
    _run_tests()

File: src/content/docs/topics/cs/coding-problems/stack/907-sum-of-subarray-minimums.go
package main

import "fmt"

func assert(condition bool, msgs ...string) {
	if !condition {
		msg := "assertion failed"
		if len(msgs) > 0 {
			msg = msgs[0]
		}
		panic(msg)
	}
}

func sumSubarrayMins(arr []int) int {
	// TODO: implement
	return 0
}

func runTests() {
	assert(sumSubarrayMins([]int{3, 1, 2, 4}) == 17)
	assert(sumSubarrayMins([]int{11, 81, 94, 43, 3}) == 444)
	assert(sumSubarrayMins([]int{3}) == 3)
	fmt.Println("all tests pass")
}

func main() { runTests() }
